File: Array/Python/Challenge/Saya.py
# Kode ini dibagikan oleh Dedad Fajar dan dapat diakses melalui github: https://github.com/VictoriaEstara/perancangan-struktur-data

class Mahasiswa:
    def __init__(self, nim, nama, umur, jml_saudara):
        self.nim = nim
        self.nama = nama
        self.umur = umur
        self.jml_saudara = jml_saudara

mahasiswa_list = []

def input_data_mahasiswa():
    nim = input("NIM : ")
    nama = input("Nama : ")
    umur = input("Umur : ")
    jml_saudara = input("Jumlah Saudara : ")
    mahasiswa = Mahasiswa(nim, nama, umur, jml_saudara)
    mahasiswa_list.append(mahasiswa)
    print("\nData mahasiswa berhasil disimpan.\n")

def tampil_data_mahasiswa():
    if not mahasiswa_list:
        print("Belum ada data mahasiswa yang tersimpan.\n")
        return
    print("\nData Mahasiswa")
    print("----------------------")
    for idx, mahasiswa in enumerate(mahasiswa_list, 1):
        print(f"{idx}. NIM : {mahasiswa.nim} Nama : {mahasiswa.nama} Umur : {mahasiswa.umur} Jumlah Saudara : {mahasiswa.jml_saudara}")
    print()

def main():
    while True:
        print("Menu Utama")
        print("----------------------")
        print("1. Input Data Mahasiswa")
        print("2. Menampilkan Data Mahasiswa")
        print("3. Exit")
        choice = input("\nPilihlah Menu Berikut : ")
        
        if choice == '1':
            print("\nInput Data Mahasiswa")
            print("----------------------")
            input_data_mahasiswa()
        elif choice == '2':
            tampil_data_mahasiswa()
        elif choice == '3':
            print("Terima kasih telah menggunakan program ini.")
            break
        else:
            print("Pilihan tidak valid. Silakan pilih menu yang tersedia.\n")

if __name__ == "__main__":
    main()

File: Array/Java/Challenge/Saya.java
// Kode ini dibagikan oleh Dedad Fajar dan dapat diakses melalui github: https://github.com/VictoriaEstara/perancangan-struktur-data

import java.util.Scanner;

class Mahasiswa {
    String nim;
    String nama;
    int umur;
    int jumlahSaudara;

    Mahasiswa(String nim, String nama, int umur, int jumlahSaudara) {
        this.nim = nim;
        this.nama = nama;
        this.umur = umur;
        this.jumlahSaudara = jumlahSaudara;
    }

    @Override
    public String toString() {
        return "NIM : " + nim + " Nama : " + nama + " Umur : " + umur + " Jumlah Saudara : " + jumlahSaudara;
    }
}

public class Saya {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Mahasiswa[] mahasiswas = new Mahasiswa[3];
        int jumlahMahasiswa = 0;

        while (true) {
            System.out.println("\nMenu Utama");
            System.out.println("----------------------");
            System.out.println("1. Input Data Mahasiswa");
            System.out.println("2. Menampilkan Data Mahasiswa");
            System.out.println("3. Exit");
            System.out.print("\nPilihlah Menu Berikut : ");
            int pilihan = scanner.nextInt();
            scanner.nextLine(); // Membuang newline character

            switch (pilihan) {
                case 1:
                    System.out.println("\nInput Data Mahasiswa");
                    System.out.println("----------------------");
                    System.out.print("NIM : ");
                    String nim = scanner.nextLine();
                    System.out.print("Nama : ");
                    String nama = scanner.nextLine();
                    System.out.print("Umur : ");
                    int umur = scanner.nextInt();
                    System.out.print("Jumlah Saudara : ");
                    int jumlahSaudara = scanner.nextInt();

                    mahasiswas[jumlahMahasiswa] = new Mahasiswa(nim , nama, umur, jumlahSaudara);
                    jumlahMahasiswa++;
                    break;
                case 2:
                    System.out.println("\nData Mahasiswa");
                    System.out.println("----------------------");
                    for (int i = 0; i < jumlahMahasiswa; i++) {
                        System.out.println((i+1) + ". " + mahasiswas[i]);
                    }
                    break;
                case 3:
                    System.out.println("Sekian dan Terima kasih!");
                    System.exit(0);
                default:
                    System.out.println("Pilihan tidak valid!");
            }
        }
    }
}